src/Microsoft.SqlTools.ServiceLayer/DacFx/ExportOperation.cs

```diff
@@ -0,0 +1,51 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlServer.Dac;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.Utility;
+using System;
+using System.Data.SqlClient;
+using System.Diagnostics;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent an in-progress export operation
+    /// </summary>
+    class ExportOperation : DacFxOperation
+    {
+        public ExportParams Parameters { get; }
+
+        private SqlConnection sqlConnection{ get; set; }
+
+        public ExportOperation(ExportParams parameters, SqlConnection sqlConnection)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            Validate.IsNotNull("sqlConnection", sqlConnection);
+            this.Parameters = parameters;
+            this.sqlConnection = sqlConnection;
+        }
+
+        public override void Execute(TaskExecutionMode mode)
+        {
+            if (this.CancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException(this.CancellationToken);
+            }
+
+            try
+            {
+                DacServices ds = new DacServices(this.sqlConnection.ConnectionString);
```

Comment:
Can we use the constructor that takes a SecureString?

Reply:
Applies to entire PR where we instantiate DacServices.

Reply:
Currently, the SqlCredential with the SecureString password doesn't get populated when the ConnectionService opens the connection.